plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConnectorModule.java

```diff
@@ -67,6 +79,39 @@ public static HeaderProvider createHeaderProvider(NodeManager nodeManager)
         {
             return FixedHeaderProvider.create("user-agent", "Trino/" + nodeManager.getCurrentNode().getVersion());
         }
+
+        /**
+         * Apache Arrow requires reflective access to certain Java internals prohibited in Java 17.
+         * Adds an error to the {@code binder} if required --add-opens is not passed to the JVM.
+         */
+        private static void verifyPackageAccessAllowed(Binder binder)
+        {
+            if (Runtime.version().compareToIgnoreOptional(Runtime.Version.parse("17")) < 0) {
```

Comment:
@hashhar Why do we need this check since Trino is only JDK 17+?

Reply:
Good catch. Thanks.

Reply:
https://github.com/trinodb/trino/pull/15725